task-3.php

```diff
@@ -0,0 +1,354 @@
+<?php
+/**
+ * @param $a
+ *
+ * @return string
+ */
+function planeSeat( $a ) {
+
+	$planeSeatManager = new SeatManager();
+	$planeSeatChecker = new SeatChecker();
+	$planeSeatChecker->setSeatManager( $planeSeatManager );
+
+	return $planeSeatChecker->init( $a );
+}
+
+/**
+ * Class SeatChecker
+ */
+class SeatChecker {
+
+	/**
+	 * Seat Manager.
+	 *
+	 * @var SeatManagerInterface
+	 */
+	protected $seatManager;
+
+	/**
+	 * Seat Place.
+	 *
+	 * @var string
+	 */
+	protected $seatPlace;
+
+	/**
+	 * Seat number.
+	 *
+	 * @var string
+	 */
+	protected $seatNumber;
+
+	/**
+	 * Seat letter.
+	 *
+	 * @var string
+	 */
+	protected $seatLetter;
+
+	/**
+	 * SeatPlace.
+	 *
+	 * @param $seatPlace
+	 *
+	 * @return string
+	 */
+	public function init( $seatPlace ) {
+		$this->seatPlace = $seatPlace;
+
+		if ( !$this->checkPlaceByNumber() || !$this->checkPlaceByLetter() ) {
+			return 'No Seat!!';
+		}
+
+		return $this->getSeatNumber() . '-' . $this->getSeatLetter();
+	}
+
+	/**
+	 * GetSeatLetter.
+	 *
+	 * @return mixed
+	 */
+	protected function getSeatLetter() {
+		return $this->seatLetter;
+	}
+
+	/**
+	 * GetSeatNumber.
+	 *
+	 * @return mixed
+	 */
+	protected function getSeatNumber() {
+		return $this->seatNumber;
+	}
+
+	/**
+	 * SetSeatLetter.
+	 *
+	 * @param $seatLetter
+	 */
+	protected function setSeatLetter( $seatLetter ) {
+		$this->seatLetter = $seatLetter;
+	}
+
+	/**
+	 * SetSeatNumber.
+	 *
+	 * @param $seatNumber
+	 */
+	protected function setSeatNumber( $seatNumber ) {
+		$this->seatNumber = $seatNumber;
+	}
+
+	/**
+	 * SetSeatManager.
+	 *
+	 * @param SeatManagerInterface $seatManager
+	 */
+	public function setSeatManager( SeatManagerInterface $seatManager ) {
+		$this->seatManager = $seatManager;
+	}
+
+	/**
+	 * GetSeatPlace.
+	 *
+	 * @return mixed
+	 */
+	public function getSeatPlace() {
+		return $this->seatPlace;
+	}
+
+	/**
+	 * checkPlaceByLetter.
+	 *
+	 * @return bool
+	 */
+	public function checkPlaceByLetter() {
+		$seat_letter = $this->getSeatPlace();
+		$arraySearch = [
+			$this->seatManager->getSeatLeftLetter(),
+			$this->seatManager->getSeatMiddleLetter(),
+			$this->seatManager->getSeatRightLetter(),
+		];
+
+		foreach ( $arraySearch as $keyLetter => $letter ) {
+			foreach ( $letter as $key => $value ) {
+
+				$resultCompareByLetter = $this->compareByLetter( $seat_letter, $value );
+				if ( $resultCompareByLetter ) {
+					$this->setSeatLetter( $key );
+
+					return true;
+				}
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * CompareByLetter.
+	 *
+	 * @param $seatLetter
+	 * @param $letterCompare
+	 *
+	 * @return bool
+	 */
+	public function compareByLetter( $seatLetter, $letterCompare ) {
+		return strpos( $seatLetter, $letterCompare[0] ) > 0 || strpos( $seatLetter, $letterCompare[1] ) > 0 ? true : false;
```

Comment:
This only checks first and last letters ('A', 'C') and not letters inside ('B').
You can make string comparison.